mlir/docs/DefiningDialects/AttributesAndTypes.md

```diff
@@ -1204,3 +1204,52 @@ void MyDialect::initialize() {
   >();
 }
 ```
+
+### Attribute / Type Constraints
```

Comment:
Can you just break this out into a new `Constraints.md` file in the same directory? We have docs on these spread out, the original comment I made directed to the partial docs we have here:  https://mlir.llvm.org/docs/DefiningDialects/Operations/#operation-arguments

Would be better to just start a new doc with these contents and start populating more content into that.